cmd/main.go

```diff
@@ -145,6 +149,20 @@ Formats:
     standard-quiet           standard go test format
     standard-verbose         standard go test -v format
 
+Format icons:
+    default                  the original unicode (✓, ∅, ✖)
+    alternative              an alternative unicode (✓, ‐, ✗)
+    hivis                    higher visibility unicode (✅, ➖, ❌)
+    text                     simple text characters (PASS, SKIP, FAIL)
+  * nerdfonts                requires a font from https://www.nerdfonts.com/
+    codicons                   
+    codicons-circle            
+    octicons                   
+    octicons-circle            
+    octicons-filled            
+    emoticons-circle         󰇵 󰇶 󰇸
+    emoticons-filled         󰱱 󰱴 󰱶
```

Comment:
Adding `text` sounds great! 

I'm concerned now that maybe we have too many options, at least for the initial release. It can be a lot to choose from, and some of them don't look all that different. `default`, `hivis`, `text` are somewhat distinct from each other.

`alternative` isn't much different from `default` and `hivis`, maybe we could leave that one out for now?

For the `nerdfonts`, maybe we could pick just one of each of the `codicons`, `octicons` and `emoticons` variants? That would give us 6 options in total. We can always add more later.

If you had to pick between the default, `-circle`, and `-filled` variants, which do you think we should take? Maybe  `codicons`, `octicons-circle`, and `emoticons-filled` ?

Reply:
Okay, will pick out some favorites. Possibly rename the finalists.

Reply:
![gotestsum-icons](https://github.com/gotestyourself/gotestsum/assets/10364051/8ce0f8a3-6c09-4f3f-8789-3ae4cf48a52f)

I like the codicons because those are the ones I started the PR with, the octicons because they are consistent between the "circle"/outline and the "filled", and the emoticons because they are fun/colorful and whimsical.
The ones that I left out are either too similar to the default ones and not worth the hassle, or they have poor usability and visibility since the color is too vague or the form is not distinct. Added unicode from go-test-report.

Will use some better names, without dashes.

Reply:
The idea was that the icons would be set in environment (or configuration?), and wouldn't be in the way.

Reply:
Reduced it down to 4, can add text and emoticons later.

Reply:
Might want to add some kind of user configuration to gotestsum, so that they can set their own icons.